src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -2446,8 +2446,18 @@ void C2_MacroAssembler::string_compare_v(Register str1, Register str2, Register
   mv(cnt2, cnt1);
   bind(L);
 
+  // We focus on the optimization of small sized string.
+  // Please check below document for string size distribution statistics.
+  // https://cr.openjdk.org/~shade/density/string-density-report.pdf
   if (str1_isL == str2_isL) { // LL or UU
-    element_compare(str1, str2, zr, cnt2, tmp1, tmp2, v4, v8, v4, encLL, DIFFERENCE, false);
+    // Below construction of v regs and lmul is based on test on 2 different boards,
+    // vlen == 128 and vlen == 256 respectively.
+    if (!str1_isL && MaxVectorSize == 16) { // UU
```

Comment:
Seems more readable to check `!encLL` instead of `!str1_isL`. As you see, we also pass this `encLL` to `element_compare` to indicate whether they are Latin strings.

Reply:
fixed.